src/app/channels/channels.component.ts

```diff
@@ -29,15 +31,16 @@ export class ChannelsComponent implements OnInit {
     private channelService: ChannelService,
     private userService: UserService,
     private route: ActivatedRoute,
-    private searchService: SearchService) { }
+    private searchService: SearchService,
+    private channelPermissions: ChannelPermissionsService) { }
 
   ngOnInit() {
     if (this.isSearch) {
       this.subscribeToSearch();
       this.loadSearchedChannelsAmount();
     } else {
       if (this.isUserChannels) {
-        this.channelFilter.user = this.userService.getUser();
+        this.channelFilter.user = this.userService.currentUser.id;
```

Comment:
This check happens twice - here and in the url subscription.
If the url contains `/user` then this is user's channel.